Add an algorithm to sniff for WebM

padenot opened the pull request:
This algorithm allows to sniff specifically for WebM, and not for an EBML document (which was effectively what was happening with just matching the first four bytes).

Now a couple questions:
- The sniffing algorithm uses bitwise operations. I most of the operation needed are defined in the `Encoding` text. Should I mention the fact that we use the same definition here ?
- This needs the ~ (bitwise inversion) operator, that is not in `Encoding`. Should I add it to `Encoding`, or just add a definition in `mimesniff` ?

This implementation is derived from <https://github.com/padenot/webm-sniff/> (that contains test vectors that have been gathered in the field and/or produced using a variety of encoders from different vendors), that is in turn derived from <https://github.com/kinetiknz/nestegg>, which is the WebM demuxer of Gecko.

padenot commented:
Hrm, it's unclear why another commit slipped in this PR, sorry.

annevk commented:
Can you rebase and force push so it only contains a single commit?

annevk commented on Overview.src.html:
```diff
@@ -1643,7 +1626,100 @@ <h2 class="no-num no-toc">Table of contents</h2>
+   Let <var>sequence</var> be the <span>byte sequence</span> to be matched,
+   where <var>sequence</var>[<var>s</var>] is <span>byte</span> <var>s</var>
+   in <var>sequence</var> and <var>sequence</var>[0] is the first
+   <span>byte</span> in <var>sequence</var>.
```
This needs out-of-bound checks, too, no?

annevk commented on Overview.src.html:
```diff
@@ -1643,7 +1626,100 @@ <h2 class="no-num no-toc">Table of contents</h2>
+  <li>
+   If the four <span title=byte>bytes</span> from
+   <var>sequence</var>[0] to <var>sequence</var>[3], are not equal to 0x1A
+   0x45 0xDF 0xA3, return false.
```
This needs out-of-bound checks, too, no?

padenot commented:
Anne, I've addressed the comment about the range check in another commit for the ease of review, I can squash those before this gets merged.

Also, this is rebased.

annevk commented:
Thank you, so the main problem here is that @GPHemsley has sorta stepped down as editor. Given that, I think I'd be okay with you squashing these commits and me merging them.

I'll ask on #whatwg if anyone has objections to that to be sure.

padenot commented:
Anne, this has been squashed and rebased.

annevk commented:
Thank you, landed as https://github.com/whatwg/mimesniff/commit/b03894eea2955bddb2202e29ee185e7882e178ab.
